cypress/integration/Buy.spec.js

```diff
@@ -0,0 +1,295 @@
+
+      summary.expectProxyBeingCreated();
+      summary.expectBought(willReceive, to);
+      summary.expectSold(willPay, from);
+      summary.expectPriceOf(price)
+    });
+
+    it('with proxy', () => {
+      const to = 'DAI';
+      const from = 'ETH';
+      const willReceive = '100';
+
+      const trade = new Trade().buy(to)(willReceive);
+
+      trade
+        .acceptTerms()
+        .execute();
+
+      nextTrade();
+
+      const willReceiveMore = '500';
+      const willPay = '1.80649';
+      const price = '276.77929 ETH/DAI';
+
+      const secondTrade = new Trade().buy(to)(willReceiveMore);
+
+      expect(secondTrade).to.pay(willPay);
+
+      const finalization = trade
+        .acceptTerms()
+        .execute();
+
+      const summary = finalization
+        .shouldNotCreateProxy()
+        .shouldCommitATrade(willPay, from, willReceiveMore, to);
+
+      summary.expectProxyNotBeingCreated();
+      summary.expectBought(willReceiveMore, to);
+      summary.expectSold(willPay, from);
+      summary.expectPriceOf(price)
+    });
+  });
+
+  context('ERC20 for ETH', () => {
+    it('without proxy and allowance', () => {
+      const from = 'DAI';
+      const to = 'ETH';
+      const willPay = '37.25778';
+      const willReceive = '0.12378';
+      const price = '301 ETH/DAI';
+
+      const trade = new Trade().buy(to)(willReceive);
+
+      expect(trade).to.pay(`${willPay}`);
+
+      const finalization = trade
+        .acceptTerms()
+        .execute();
+
+      finalization.shouldCreateProxy();
+      expect(finalization.currentTx).to.succeed();
+
+      finalization.shouldSetAllowanceFor(from);
+      expect(finalization.currentTx).to.succeed();
+
+      const summary = finalization.shouldCommitATrade(willPay, from, willReceive, to);
+
+      summary.expectBought(willReceive, to);
+      summary.expectSold(willPay, from);
+      summary.expectPriceOf(price);
+    });
+
+    it("with proxy and no allowance", () => {
+      const from = 'ETH';
+      const to = 'DAI';
+      const willPay = '1';
+
+      const trade = new Trade()
+        .sell(from)(willPay)
+        .buy(to)()
+        .acceptTerms()
+        .execute();
+
+      nextTrade();
+
+      const nextFrom = 'DAI';
+      const nextTo = 'ETH';
+      const nextWillPay = '37.25778';
+      const nextWillReceive = '0.12378';
+      const price = '301 ETH/DAI';
+
+      const secondTrade = new Trade()
+        .sell(nextFrom)()
+        .buy(nextTo)(nextWillReceive);
+
+      expect(trade).to.receive(`${nextWillReceive}`);
+
+      const nextFinalization = secondTrade
+        .acceptTerms()
+        .execute();
+
+      nextFinalization
+        .shouldNotCreateProxy()
+        .shouldSetAllowanceFor(nextFrom);
+
+      expect(nextFinalization.currentTx).to.succeed();
+
+      const finalSummary = nextFinalization
+        .shouldCommitATrade(nextWillPay, nextFrom, nextWillReceive, nextTo);
+
+      finalSummary.expectProxyNotBeingCreated();
```

Comment:
and here. Let me know if i get this right: first is "in progress" view and latter is summary view.